include/aws/http/private/no_proxy.h

```diff
@@ -0,0 +1,27 @@
+#ifndef AWS_NO_PROXY_H
+#define AWS_NO_PROXY_H
+
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/http/http.h>
+AWS_PUSH_SANE_WARNING_LEVEL
+AWS_EXTERN_C_BEGIN
+
+/*
+ * Check if a host should bypass the proxy based on the NO_PROXY environment variable.
+ * Since NO_PROXY has no standard yet. Follows the curl implementation from noproxy.c at 8.14.1.
+ * https://github.com/curl/curl/blob/curl-8_14_1/lib/noproxy.c
+ *
+ * NO_PROXY is a comma-separated list of domain names, hostnames, or IP addresses that
+ * should bypass the proxy.
+ *
+ * Returns true if the host should bypass the proxy.
+ */
+AWS_HTTP_API bool aws_check_no_proxy(struct aws_allocator *allocator, struct aws_byte_cursor host);
```

Comment:
from testing perspective should this just take a no_proxy string to match against and have some other code retrieve no_proxy from env?

Reply:
also the name could be better. maybe something like aws_http_host_matches_no_proxy?